### PR TITLE
`Development`: Refactor ExerciseDateService to reduce service dependencies

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context
Makes the instance method `ExerciseDateService::getDueDate` static to reduce dependencies on that service and to make it usable from static contexts in other services. This reduces code duplication (as noticed in #5833).

### Steps for Testing

1. Code review should be enough, no logic has been changed.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
unchanged

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java [hunk 1]
```diff
@@ -303,6 +292,21 @@ private static void validateTimeOfComplaintOrRequestMoreFeedback(Result result,
...
+        possibleComplaintStartDates.add(result.getCompletionDate());
+
+        final Optional<ZonedDateTime> relevantDueDate = Optional.ofNullable(studentParticipation).flatMap(ExerciseDateService::getDueDate);
+        relevantDueDate.ifPresent(possibleComplaintStartDates::add);
+
+        if (exercise.getAssessmentDueDate() != null) {
+            possibleComplaintStartDates.add(exercise.getAssessmentDueDate());
+        }
+
+        // At least result.getCompletionDate is present, since it was checked earlier
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java:306]: Maybe add this as javadoc above the method

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Changes look good. Left one comment.
If you want to, you can think about adding a static import for that method, if it is used multiple times in a class

**Review by JohannesStoehr — APPROVED** [review]
Code changes look good to me

**Review by frederik-kt — APPROVED** [review]
Code changes look good.

**Pull request merged**